src/browser/base/content/zen-styles/zen-tabs/vertical-tabs.css

```diff
@@ -503,28 +598,38 @@
         /*animation: zen-vtabs-animation 0.3s ease-in-out;*/
         -moz-window-dragging: no-drag;
         overflow: hidden;
-        transition: width .2s !important;
+        /* Opening transition */
+        transition: 
+          width var(--tabs-expanding-time) var(--tabs-expanding-timing),
+          border-top-color var(--tabs-expanding-time) var(--tabs-expanding-timing),
+          border-right-color var(--tabs-expanding-time) var(--tabs-expanding-timing),
+          /* Show background-color faster, otherwise, background content can be seen */
+          background-color var(--tabs-appearing-color-time) cubic-bezier( 0, 0.99, 0, 1 ) !important;
 
         & .tab-throbber,
         & .tab-icon-pending,
         & .tab-icon-image,
         & .tab-sharing-icon-overlay,
         & .tab-icon-overlay {
-          transition: 0.1s !important;
+          transition: var(--tabs-expanding-time) !important;
         }
 
+        @media ((-moz-bool-pref: 'zen.view.compact') and (-moz-bool-pref: 'zen.view.compact.hide-toolbar')) {
+          margin-top: calc(var(--zen-toolbox-padding) + 7px);
+          padding-bottom: calc(var(--zen-toolbox-padding) + 7px);
+        }
       }
 
       /* Make pinned tabs stay in a single line */
-      #vertical-pinned-tabs-container {
+      #vertical-pinned-tabs-container:has(tab:not([hidden])) {
         display: flex;
         flex-direction: column;
         gap: 0 !important;
 
         margin-bottom: 8px;
         position: relative;
   
-        &:has(tab:not([hidden]))::after {
+        &::after {
```

Comment:
Why remove this? It's supposed to be only a separator when there's pinned tabs

Reply:
I just moved up, because when there wore no pinned tabs, the 8px margin-bottom wore still being applied causing an alignment difference between collapsed and expanded.

```diff
- #vertical-pinned-tabs-container {
+ #vertical-pinned-tabs-container:has(tab:not([hidden])) {
```

Now when there is no pinned tab, there will be no separator nor margin